tests/security/audit/ausearch.pm

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-Later
+#
+# Summary: Verify the "ausearch" utility can search the audit log file for certain events using various keys or
+#          other characteristics of the logged format
+# Maintainer: llzhao <llzhao@suse.com>, shawnhao <weixuan.hao@suse.com>
+# Tags: poo#81772
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $tmp_output = '/tmp/out';
+    my $tmp_backup = '/tmp/backup';
+
+    select_console 'root-console';
+
+    # Make sure audit service is started
+    assert_script_run('systemctl start auditd');
+
+    # Generate audit records for testing
+    assert_script_run('systemctl stop apparmor');
+    assert_script_run('systemctl start apparmor');
+
+    # Search for an event based on the given filename
+    assert_script_run("ausearch -f /etc > $tmp_output");
+
+    # Extract pid from output log
+    script_run("tail -1 $tmp_output > $tmp_backup");
+    script_run("cat $tmp_backup | cut -d '=' -f 9 > $tmp_output");
+    my $pid = script_output("cat $tmp_output | cut -d ' ' -f 1");
+
+    # Search for an event matching the given process ID
+    assert_script_run("ausearch -p $pid > $tmp_output");
+
+    # Extract event id from output log
+    script_run("tail -1 $tmp_output > $tmp_backup");
+    script_run("awk -F: '{print \$2}' $tmp_backup > $tmp_output");
+    my $event_id = script_output("cat $tmp_output | cut -d ')' -f 1");
```

Comment:
Similar comment as above

Reply:
I didn't quite understand your comment here for line 39-42. Since this is to extract an event id from audit log. Could you explain further?

Reply:
One line code should cover this function :)
Anyway, not a big deal, you can keep the current stytle